tests/gtests/sw_testhelpers.cc

```diff
@@ -43,6 +43,10 @@ void create_test_soillayers(unsigned int nlayers,
       "requested number of soil layers (n = %d) is not accepted.\n", nlayers);
   }
 
+  if(LogInfo->stopRun) {
+    sw_check_exit(swFALSE, LogInfo);

```

Comment:
In `create_test_soillayers()` for unit tests, why `sw_check_exit(swFALSE, LogInfo);` instead of recoiling early as elsewhere?